### PR TITLE
8234 Renamed command to start with Appium prefix, changed order of some parameters, added option to find elements by pure xpath

### PR DESCRIPTION
**Pull request by przegwar**

### CODE REVIEW

**Review by DanielWilczek — APPROVED** [review]

File: G1ANT.Appium/ElementHelper.cs [hunk 1]
```diff
@@ -83,5 +85,20 @@ private static AndroidElement GetElementByText(string text)
                 throw new ArgumentException($"Element with provided text was not found.");
             }
         }
+
+        private static AndroidElement GetElementByXpath(string xpath)
+        {
+            var driver = OpenCommand.GetDriver();
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (DanielWilczek) [inline G1ANT.Appium/ElementHelper.cs]: var

File: G1ANT.Appium/ElementsHelper.cs [hunk 2]
```diff
@@ -84,5 +86,20 @@ private static List<AndroidElement> GetElementsByText(string text)
                 throw new ArgumentException($"Element with provided text was not found.");
             }
         }
+
+        private static List<AndroidElement> GetElementsByXpath(string xpath)
+        {
+            var driver = OpenCommand.GetDriver();
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (DanielWilczek) [inline G1ANT.Appium/ElementsHelper.cs]: var

**Review by DanielWilczek — APPROVED** [review]

**Comment by Vegebutcher** [comment]
Please rename the `Name` argument to `Search`. It should also be placed as the first argument, e.g.: `appium.click search ... by ...`, although it's not that important as the interpreter doesn't care about the argument order.

File: G1ANT.Appium/Commands/AppiumClickCommand.cs [hunk 3]
```diff
@@ -28,9 +28,9 @@ public void Execute(Arguments arguments)
 
             if(by == SearchBy.XY)
             {
-                TouchAction a2 = new TouchAction(OpenCommand.GetDriver());
+                TouchAction clickAction = new TouchAction(OpenCommand.GetDriver());
```
Comment (MarekKosinski) [inline G1ANT.Appium/Commands/AppiumClickCommand.cs:31]: var **touchAction**

File: G1ANT.Appium/Commands/AppiumGetTextCommand.cs [hunk 4]
```diff
@@ -9,12 +9,12 @@ public class GetTextCommand : Language.Command
     {
         public class Arguments : CommandArguments
         {
-            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
-            public TextStructure By { get; set; } = new TextStructure("");
-
             [Argument(Tooltip = "Provide name of the capaility")]
             public TextStructure Name { get; set; } = new TextStructure("");
 
+            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
```
Comment (MarekKosinski) [inline G1ANT.Appium/Commands/AppiumGetTextCommand.cs:15]: wich -> **which**

File: G1ANT.Appium/Commands/AppiumGetTextFromElementsCommand.cs [hunk 5]
```diff
@@ -9,12 +9,12 @@ public class GetTextFromElementsCommand : Language.Command
     {
         public class Arguments : CommandArguments
         {
-            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
-            public TextStructure By { get; set; } = new TextStructure("");
-
             [Argument(Tooltip = "Provide name of the capaility")]
             public TextStructure Name { get; set; } = new TextStructure("");
 
+            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
```
Comment (MarekKosinski) [inline G1ANT.Appium/Commands/AppiumGetTextFromElementsCommand.cs:15]: wich -> **which**

File: G1ANT.Appium/Commands/AppiumGetTextFromElementsCommand.cs [hunk 6]
```diff
@@ -9,12 +9,12 @@ public class GetTextFromElementsCommand : Language.Command
     {
         public class Arguments : CommandArguments
         {
-            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
-            public TextStructure By { get; set; } = new TextStructure("");
-
             [Argument(Tooltip = "Provide name of the capaility")]
```
Comment (MarekKosinski) [inline G1ANT.Appium/Commands/AppiumGetTextFromElementsCommand.cs:12]: Those arguments are used twice here and in "G1ANT.Appium/Commands/GetTextCommand.cs → G1ANT.Appium/Commands/AppiumGetTextCommand.cs"

Extract them to a base class.

File: G1ANT.Appium/Commands/AppiumScrollCommand.cs [hunk 7]
```diff
@@ -13,12 +13,12 @@ public class ScrollCommand : Language.Command
     {
         public class Arguments : CommandArguments
         {
-            [Argument(Tooltip = "Specify by wich capability the element will be searched")]
-            public TextStructure By { get; set; } = new TextStructure("");
-
             [Argument(Tooltip = "Provide name of the capaility")]
```
Comment (MarekKosinski) [inline G1ANT.Appium/Commands/AppiumScrollCommand.cs:16]: Same arguments as in "G1ANT.Appium/Commands/PresentCommand.cs → G1ANT.Appium/Commands/AppiumPresentCommand.cs"

Extract them to a base class.

File: G1ANT.Appium/ElementHelper.cs [hunk 8]
```diff
@@ -27,7 +29,8 @@ public static AndroidElement GetElement(SearchBy by, string name)
         private static AndroidElement GetElementById(string id)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementHelper.cs:32]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementHelper.cs [hunk 9]
```diff
@@ -42,7 +45,8 @@ private static AndroidElement GetElementById(string id)
         private static AndroidElement GetElementByAccessibilityId(string accessibilityId)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementHelper.cs:48]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementHelper.cs [hunk 10]
```diff
@@ -57,7 +61,8 @@ private static AndroidElement GetElementByAccessibilityId(string accessibilityId
         private static AndroidElement GetElementByPartialId(string partialId)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementHelper.cs:64]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementHelper.cs [hunk 11]
```diff
@@ -72,7 +77,8 @@ private static AndroidElement GetElementByPartialId(string partialId)
         private static AndroidElement GetElementByText(string text)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementHelper.cs:80]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementHelper.cs [hunk 12]
```diff
@@ -83,5 +89,21 @@ private static AndroidElement GetElementByText(string text)
                 throw new ArgumentException($"Element with provided text was not found.");
             }
         }
+
+        private static AndroidElement GetElementByXpath(string xpath)
+        {
+            var driver = OpenCommand.GetDriver();
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementHelper.cs:96]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementsHelper.cs [hunk 13]
```diff
@@ -43,7 +46,8 @@ private static List<AndroidElement> GetElementsById(string id)
         private static List<AndroidElement> GetElementsByAccessibilityId(string accessibilityId)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementsHelper.cs:49]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementsHelper.cs [hunk 14]
```diff
@@ -58,7 +62,8 @@ private static List<AndroidElement> GetElementsByAccessibilityId(string accessib
         private static List<AndroidElement> GetElementsByPartialId(string partialId)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementsHelper.cs:65]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementsHelper.cs [hunk 15]
```diff
@@ -73,7 +78,8 @@ private static List<AndroidElement> GetElementsByPartialId(string partialId)
         private static List<AndroidElement> GetElementsByText(string text)
         {
             var driver = OpenCommand.GetDriver();
-            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementsHelper.cs:81]: magic number. Move it to a const with proper name.

File: G1ANT.Appium/ElementsHelper.cs [hunk 16]
```diff
@@ -84,5 +90,21 @@ private static List<AndroidElement> GetElementsByText(string text)
                 throw new ArgumentException($"Element with provided text was not found.");
             }
         }
+
+        private static List<AndroidElement> GetElementsByXpath(string xpath)
+        {
+            var driver = OpenCommand.GetDriver();
+            var wait = new WebDriverWait(driver, TimeSpan.FromSeconds(7));
```
Comment (MarekKosinski) [inline G1ANT.Appium/ElementsHelper.cs:97]: magic number. Move it to a const with proper name.

**Review by MarekKosinski — APPROVED** [review]

**Pull request merged**